polkaj-scale/src/main/java/io/emeraldpay/polkaj/scale/reader/UInt128Reader.java

```diff
@@ -5,23 +5,14 @@
 
 import java.math.BigInteger;
 
-public class UInt128Reader implements ScaleReader<BigInteger> {
+public class UInt128Reader extends BigIntReader implements ScaleReader<BigInteger> {
 
-    public static final int SIZE_BYTES = 16;
-
-    public static void reverse(byte[] value) {
-        for (int i = 0; i < value.length / 2; i++) {
-            int other = value.length - i - 1;
-            byte tmp = value[other];
-            value[other] = value[i];
-            value[i] = tmp;
-        }
-    }
+    BigIntReader bir = new BigIntReader();
 
     @Override
     public BigInteger read(ScaleCodecReader rdr) {
-        byte[] value = rdr.readByteArray(SIZE_BYTES);
-        reverse(value);
+        byte[] value = rdr.readByteArray(bir.UINT128_SIZE_BYTES);
+        bir.reverse(value);
         return new BigInteger(1, value);
     }
```

Comment:
I'm only seeing a small slice of the project, so might be totally wrong on this one. Could you have the base class BigIntReader implement the ScaleReader and move the read method override there to reuse less code?

Reply:
I think I'll leave it as is for now (unless Peter thinks it's a good idea to change it) since when it comes to implementing these classes in our project, it's more intuitive to call the UInt64Reader/Writer and UInt128Reader/Writer separately